ivy_tests/test_ivy/test_frontends/test_numpy/test_fft/test_discrete_fourier_transform.py

```diff
@@ -138,3 +138,29 @@ def test_numpy_fftfreq(n, sample_rate, frontend, test_flags, fn_tree, on_device)
         n=n,
         d=d,
     )
+
+@handle_frontend_test(
+    fn_tree="numpy.fft.fft2",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float_and_complex"),
+        shape=(2, 2),
+        array_api_dtypes=True
+    ),
+    s=st.sampled_from([(2, 2), (3, 3), (4, 4)]),
+    axes=st.sampled_from([(0, 1), (-2, -1)]),
+    norm=st.sampled_from([None, "backward", "ortho", "forward"]),
+)
+def test_numpy_fft2(dtype_and_x, s, axes, norm, frontend, test_flags, fn_tree, on_device):
+    input_dtype, a = dtype_and_x
+    helpers.test_frontend_function(
```

Comment:
Seems like the test failed for Tensorflow, do you mind checking back?

Reply:
Hey, many thanks for the feedback! 
So I resolved this issue, but still failing one test (IvyBackendException). However, the test being failed is also failed by other functions within the file (namely: rfft and numpy_fft). 
As a result, I believe these errors are caused by something a bit deeper

Reply:
It's fixed in this commit e3797979, please pull the latest change.

Reply:
Just pushed another commit dd57e7d to fix `fft` and your `fft2` as well. Pull the latest change and run the tests, they should all pass now.

Reply:
ok will do, thank you :)

Reply:
Hey, so I've updated the branch and made most of the requested changes. 

Unfortunately, when running the tests, I still encounter a tensorflow error that states complex64 dtypes are incompatible with Numpy. 
From what I understnad, shouldn't this be moot since the function fft2 itself converts arrays to complex128 ivy arrays and back again. 
Can't seem to understand why this error is arising in the first place. 
Any help is greatly appreciated, many thanks :)

Reply:
Looking at your branch's [commit history](https://github.com/joshg298/ivy/commits/jjg_branch), seems like you haven't gotten my commit dd57e7d